### PR TITLE
Initial implementation of the gate evaluation command

### PR DESCRIPTION
**Pull request by andrea-mosk**
### What and why?

Initial implementation of a new "gate evaluation" command.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/gate/api.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import {getRequestBuilder} from '../../helpers/utils'
+
+import {Payload} from './interfaces'
+
+export const evaluateGateRules = (request: (args: AxiosRequestConfig) => AxiosPromise<AxiosResponse>) => async (
+  evaluateRequest: Payload,
+  write: Writable['write']
+) => {
+  // Converting the payload to the format expected from Rapid
```
Comment (juan-fernandez) [inline src/commands/gate/api.ts:13]: nitpick: I would avoid using references to services / technologies used by datadog

File: src/commands/gate/evaluate.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    // To be extended
+    try {
+      await api.evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout))
+    } catch (error) {
+      this.context.stderr.write(error)
+    }
+  }
+}
+
+// TODO commented out for now to not expose anything to customers.
```
Comment (juan-fernandez) [inline src/commands/gate/evaluate.ts:77]: with the latest change this is not needed anymore, is it?

**Review by juan-fernandez — APPROVED** [review]

File: src/commands/gate/api.ts (hunk 1 above)
Comment (andrea-mosk) [reply src/commands/gate/api.ts]: Oh, you're right, changed!

**Review by liashenko — APPROVED** [review]

**Comment by buraizu** [comment]
Created DOCS-5488 to track.

**Review by alai97 — APPROVED** [review]
Looks good for docs!

**Pull request merged**